### PR TITLE
PHP worker and improvements

### PR DESCRIPTION
**Pull request by talyguryn**
- worker for PHP events
- simplify code for simple workers (based on EventWorker with no additional logic)

### CODE REVIEW

File: lib/event-worker.ts [hunk 1]
```diff
@@ -1,10 +1,43 @@
...
+import * as WorkerNames from './workerNames';
+import { GroupWorkerTask } from 'hawk-worker-grouper/types/group-worker-task';
+import { EventWorkerTask } from './types/event-worker-task';
 
 /**
  * Defines a Worker that handles events from Catcher.
- * Used to extract Project Id from Integration Token and to provide some common methods.
- *
- * catherTypes -> 'error/*'
+ * Used to extract Project Id from Integration Token
```
Comment (neSpecc) [inline lib/event-worker.ts:8]:
```suggestion
```

File: lib/event-worker.ts [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
+ * and to provide some common methods.
```
Comment (neSpecc) [inline lib/event-worker.ts:9]:
```suggestion
 * Used for the common language-workers that require no specific logic (source maps consuming etc). Other workers can override `handle()` method 
```

File: workers/php/tests/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    type: null,
+    backtrace: null,
+    timestamp: 1564948772936,
+  },
+} as PhpEventWorkerTask;
+
+describe('PhpEventWorker', () => {
+  const worker = new PhpEventWorker();
+
+  test('should start correctly', async () => {
```
Comment (neSpecc) [inline workers/php/tests/index.test.ts:24]: unuseful test

File: workers/php/tests/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    await worker.start();
+  });
+
+  test('should have correct catcher type', () => {
+    expect(worker.type).toEqual('errors/php');
+  });
+
+  test('should not handle bad event data', async () => {
```
Comment (neSpecc) [inline workers/php/tests/index.test.ts:32]: Seems like this test is related to the EventWorker, not PhpWorker

File: workers/php/tests/index.test.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    const handleEvent = async (): Promise<void> => {
+      await worker.handle({} as PhpEventWorkerTask);
+    };
+
+    expect(handleEvent).rejects.toThrowError();
+  });
+
+  test('should handle good event data', async () => {
```
Comment (neSpecc) [inline workers/php/tests/index.test.ts:40]: Seems like this test is related to the EventWorker, not PhpWorker

File: workers/php/tests/index.test.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    await worker.handle(testEventData);
+
+    expect(mockedAmqpChannel.sendToQueue).toHaveBeenCalledTimes(1);
+  });
+
+  test('should finish correctly', async () => {
```
Comment (neSpecc) [inline workers/php/tests/index.test.ts:46]: unuseful test

File: workers/php/tests/index.test.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import PhpEventWorker from '../src';
+import { PhpEventWorkerTask } from '../types/php-event-worker-task';
+import '../../../env-test';
+import { mockedAmqpChannel } from './rabbit.mock';
+jest.mock('amqplib');
```
Comment (neSpecc) [inline workers/php/tests/index.test.ts:5]: are you sure this is necessary?

File: workers/php/tests/rabbit.mock.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import amqp from 'amqplib';
```
Comment (neSpecc) [inline workers/php/tests/rabbit.mock.ts:1]: i think this mock can be deleted at all

File: workers/php/types/php-event-worker-task.d.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import { EventWorkerTask } from '../../../lib/types/event-worker-task';
+import { DecodedEventData } from 'hawk.types';
+
+/**
+ * Describe a context passed from Php Catcher
+ */
+interface PhpEventPayload extends DecodedEventData {}
```
Comment (neSpecc) [inline workers/php/types/php-event-worker-task.d.ts:7]: pull the new version of the `hawk.types` package , please. There should be a generic with Addons. Something like that:

```ts
interface PhpEventPayload extends DecodedEventData<PhpAddons> {}
```

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: workers/php/types/php-event-worker-task.d.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import { EventWorkerTask } from '../../../lib/types/event-worker-task';
+import { DecodedEventData, EventAddons } from 'hawk.types';
+
+/**
+ * Describe a context passed from Php Catcher
+ */
+interface PhpEventPayload extends DecodedEventData<EventAddons> {}
```
Comment (neSpecc) [inline workers/php/types/php-event-worker-task.d.ts:7]: There should be `PhpAddons`. Described [there](https://github.com/codex-team/hawk.types/blob/master/src/base/event/addons/php.ts)

**Review by neSpecc — APPROVED** [review]

File: workers/python/src/index.ts [hunk 11]
```diff
@@ -14,37 +11,12 @@ export default class PythonEventWorker extends EventWorker {
...
    *
    * @param event - event to handle
    */
   public async handle(event: PythonEventWorkerTask): Promise<void> {
-    await this.addTask(WorkerNames.GROUPER, {
-      projectId: event.projectId,
-      catcherType: this.type,
-      event: event.payload,
-    } as GroupWorkerTask);
+    return super.handle(event);
```
Comment (nikmel2803) [inline workers/python/src/index.ts:20]: и тут

File: workers/nodejs/src/index.ts [hunk 12]
```diff
@@ -19,23 +17,6 @@ export default class NodeJSEventWorker extends EventWorker {
...
-  /**
-   * Validate passed event data
-   *
-   * @param {NodeJSEventWorkerTask} event - event to be validated
-   */
-  private validate(event: NodeJSEventWorkerTask): void {
-    if (!event.projectId || !event.payload) {
-      throw new Error('Bad data was given');
-    }
+    return super.handle(event);
```
[the remark this answers is not in the document]
Comment (talyguryn) [reply workers/nodejs/src/index.ts:20]: оставил эти методы только для указания особого типа переменной

image

**Review by nikmel2803 — APPROVED** [review]

**Pull request merged**